ethexe/common/src/crypto/address.rs

```diff
@@ -58,26 +59,36 @@ impl From<PublicKey> for Address {
 }
 
 impl FromStr for Address {
-    type Err = Error;
+    type Err = FromHexError;
 
-    fn from_str(s: &str) -> Result<Self> {
-        Ok(Self(crate::decode_to_array(s)?))
+    fn from_str(s: &str) -> Result<Self, FromHexError> {
+        crate::decode_to_array(s).map(Self)
     }
 }
 
+#[derive(Debug, Display)]
+#[display("{:?}", self)]
+pub enum FromActorIdError {
+    #[debug("First 12 bytes are not 0, it is not ethereum address")]
+    NotEthereumAddress,
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for FromActorIdError {}
```

Comment:
`Error` is in `core` since 1.81. you can add `derive(Error)`.

Reply:
https://docs.rs/derive_more/latest/derive_more/derive.Error.html

Reply:
Done